core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -47,6 +48,8 @@ private[spark] class CoarseGrainedExecutorBackend(
   var executor: Executor = null
   @volatile var driver: Option[RpcEndpointRef] = None
 
+  private[this] val ser: SerializerInstance = env.closureSerializer.newInstance()
```

Comment:
Done.  Even if we have to lookup the thread-local, that's probably cheaper than creating a whole new serializer for each task.